src/services/email/handlers/processEmails.tsx

```diff
@@ -0,0 +1,116 @@
+  }, {
+    "templateBase": "initial-submission-medicaid-spa-state",
+    "sendTo": ["submitterEmail"],
+  }],
+  "initial-submission-chip-spa": [{
+    "templateBase": "initial-submission-chip-spa-cms",
+    "sendTo": [process.env.chipEmail],
+    "addCC": [`${process.env.chipCCList}`],
+  }, {
+    "templateBase": "initial-submission-chip-spa-state",
+    "sendTo": ["submitterEmail"],
+  }],
+}
+
+const createSendTemplatedEmailCommand = (data) =>
+  new SendTemplatedEmailCommand({
+    Source: process.env.emailSource ?? "kgrue@fearless.tech",
+    Destination: {
+      ToAddresses: data.ToAddresses,
+      CcAddresses: data.CcAddresses,
+    },
+    TemplateData: JSON.stringify({ applicationEndpoint: "onemac.cms.gov", ...data }),
+    Template: `${data.templateBase}_${process.env.stage}`,
+    ConfigurationSetName: process.env.emailConfigSet,
+  });
+
+export const main = async (event: KafkaEvent) => {
+  console.log("Received event (stringified):", JSON.stringify(event, null, 4));
+
+  const emailQueue: any[] = [];
+  const getStateUsersFor: string[] = [];
+  Object.values(event.records).forEach((oneSource) =>
+    oneSource.forEach((encodedRecord) => {
+      if (!encodedRecord.value) return;
+      const record = { id: decode(encodedRecord.key), ...JSON.parse(decode(encodedRecord.value)) };
+      console.log("here is the decoded record: ", record);
+      if (record?.origin !== "micro") return;
+      if (!record?.actionType) record.actionType = "initial-submission";
+      record.proposedEffectiveDateNice = record?.proposedEffectiveDate ? (new Date(record.proposedEffectiveDate)).toDateString() : "Pending";
+
+      const emailsConfig = `${record.actionType}-${record.authority.replace(" ", "-")}`;
+      if (!emailsToSend[emailsConfig]) return;
+      record.territory = record.id.toString().substring(0, 2);
+      console.log("matching email config: ", emailsToSend[emailsConfig]);
+      emailsToSend[emailsConfig].map((oneEmail) => {
+        const theEmail = { ...oneEmail, ...record };
+        theEmail.ToAddresses = oneEmail.sendTo.map((oneAddress) => {
+          if (oneAddress === "submitterEmail") return `"State User Substitute" <k.grue.stateuser@gmail.com>`; //`"${theEmail.submitterName}" <${theEmail.submitterEmail}>`;
+          if (oneAddress === "allStateUsers") getStateUsersFor.push(record.territory);
+          return oneAddress;
+        });
+
+        theEmail.formattedFileList = `<ul><li>${theEmail.attachments.map((anAttachment) => anAttachment.title + ": " + anAttachment.filename).join('</li><li>')}</li></ul>`;
+        theEmail.textFileList = `${theEmail.attachments.map((anAttachment) => anAttachment.title + ": " + anAttachment.filename).join('\n')}\n\n`;
+        theEmail.ninetyDaysDateNice = theEmail?.submissionDate ? DateTime.fromMillis(theEmail.submissionDate)
+          .plus({ days: 90 })
+          .toFormat("DDDD '@ 11:59pm' ZZZZ") : "Pending";
+        emailQueue.push(theEmail);
+      })
+    }));
```

Comment:
I am interested to hear the specific flaws in the logic.  I agree the code is currently dense, but it is also on the fourth refactor and I work faster when I can see everything at once.  Because there are some aspects yet uncovered (what the final decision is for the proposedEffectiveDate and what other needs some of the other emails will have (we know at least CPOC and SRT are required)), I request that we keep this in a format I can most quickly work with until at least a few iterations.

Reply:
I disagree that your organization is cleaner and easier for me.  But I understand I am a dinosaur in some ways, so will endeavor to match the style of the team.

Reply:
oh... and I've already tried the mapAddresses to be in the forEach and the trouble comes with AllStateUsers needs to be an async function because it depends on Cognito, but MOST emails do not use AllStateUsers, so seemed like a lot of overhead for a mostly synchonous process.